pkg/route/api/allocator/allocator.go

```diff
@@ -0,0 +1,30 @@
+package allocator
```

Comment:
Ok - let me add a TODO for now. Its going to take me some time to get a handle on how to do that/how it works, so it might be safer to do it in a separate PR. 
That sound good @smarterclayton ?